include/ProgramRepresentation/DisjointPVAliasSets.h

```diff
@@ -52,6 +52,8 @@ class DisjointPVAliasSets {
     // aliases element1 and element2 by putting them into the same set. if one of
     // the elements already belongs in an existing set, both elements will be in
     // that set. otherwise, they will be put into a new set.
+    // additionally, if both sets (set1 and set2) exist, set2's set number
+    // will be changed to be the maximum of set1's set number plus 1.
```

Comment:
The sets have a maximum set number because a `PVAliasSet` can contain program variables with different set numbers. For example, if we have set `A = {a, b, c}` and set `B = {x, y, z}`, independently, the program variables in set `A` and set `B` have a set number of 0. But if they are merged, then elements `x, y, z` will be assigned a set number of 1, which is `max(set1) + 1 = 0 + 1 = 1`. I do this to ensure that merged elements are assigned a unique set number relative to the merged set

Reply:
Maybe I don't understand the semantics of the set numbers properly, but this seems wrong to me. In particular, I don't understand why in the merged set, all of the numbers for the second set are equivalent.

Continuing your example with `A` and `B`, suppose that `B` is the result of the merger of two other sets `B'` and `B''`, with the result that the set numbers for `{x, y, z}` are `{0, 0, 1}`, respectively. It seems problematic that the fact that that `B` has two components `{x, y}` (with set number 0) and `{z}` (with set number 1) will be lost using your algorithm: the merged set will give set number 1 to all of the former elements of `B`, regardless of their current set number.

Reply:
(Let's discuss this in our meeting this afternoon.)